app/src/main/java/unipd/se18/ocrcamera/BlurGalleryActivity.java

```diff
@@ -0,0 +1,143 @@
+
+    /**
+     *  Directory of the Photos
+     */
+    private final String PHOTOS_FOLDER = Environment.getExternalStoragePublicDirectory(Environment.DIRECTORY_PICTURES) + "/OCRCameraDB";
+
+
+    //private final String PHOTOS_FIX_ERROR = "/storage/emulated/0/Images/OCRCameraDB";
+    /**
+     * ListView to show all the ingredients in a listllayout
+     */
+    private ListView listView;
+
+    /**
+     * ArrayList of object (see class blur object)
+     */
+    private ArrayList<BlurObject> arrayBlur = new ArrayList<>();
+
+    /**
+     * listllayout of elements inside PHOTOS_FOLDER
+     */
+    private String fileNames[];
+
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        setContentView(R.layout.listllayout);
+        super.onCreate(savedInstanceState);
+
+        File path = new File(PHOTOS_FOLDER);
+        fileNames = path.list();
+        if (fileNames != null) {
+            setGallery();
+        }
+        else {
+            Toast.makeText(this, "No images found", Toast.LENGTH_SHORT).show();
+            this.finish();
+            //Exit from Gallery
+        }
+
+        //View preparation
+        listView = (ListView) findViewById(R.id.listview);
+        BlurObjectAdapter adapter = new BlurObjectAdapter(this, R.layout.listllayout, arrayBlur);
+        listView.setAdapter(adapter);
+        //show max blur value
+        TextView maxview = findViewById(R.id.textViewMax);
+        maxview.setText(String.valueOf(findMax(arrayBlur)));
+
+
+    }
+
+    /**
+     * Method that creates a blurObject for every image in the folder
+     *
+     */
+    private void setGallery() {
+        int count =0;
+        for (int i = 0; i < fileNames.length; i++) {                                                          //imagepath.length = number of elements in the folder
+            if (getExtension(fileNames[i]) == "jpg") {                                          //checks if it is an image
+
+                try {
+                    File f = new File(PHOTOS_FOLDER, fileNames[i]);                             //creates a new object for each element
+                    Bitmap image = BitmapFactory.decodeStream(new FileInputStream(f));
+                    BlurObject obj = new BlurObject(fileNames[i], image);
+                    arrayBlur.add(obj);
+
+                } catch (FileNotFoundException e) {
+                    e.printStackTrace();
+                    Log.e("err", "filenotfound");
+                } catch (IllegalArgumentException e) {
+                    e.printStackTrace();
+                    Log.e("err", "illegalargument");
+                }
+                count++; //check how many photos load
+                Log.e("check", Integer.toString(count));
+            }
+        }
+    }
+
+    /**
+     *  Method to show the max value of the blur
+     * @param list list of BlurObject
+     * @return double value
+     */
+    public double findMax(ArrayList<BlurObject> list){

```

Comment:
Java offer the possibility of using Collections.max, so you could avoid re-implementing this method.

Reply:
In this case i was evaluating the maximum on objects I created, so the max value is evaluated on a specific parameter of the object , i don't really think it would work.